index.js

```diff
@@ -42,17 +43,17 @@ module.exports = function (filename, opts) {
 			return;
 		}
 
-		this.push(new gutil.File({
-			cwd: firstFile.cwd,
-			base: firstFile.base,
-			path: path.join(firstFile.base, filename),
-			contents: zip.generate({
-				type: 'nodebuffer',
-				compression: opts.compress ? 'DEFLATE' : 'STORE',
-				comment: opts.comment
-			})
-		}));
-
-		cb();
+		zip.end(function () {
+			zip.outputStream.pipe(concatStream(function (data) {
```

Comment:
Yeah, makes no difference though since we're only handling one buffer at a time.